Test failure on big-endian systems

With the rebuild of fscacher on Debian Unstable/Sid ( https://tracker.debian.org/pkg/fscacher ), the following test error via Debian autopkgtest is found on s390x architecture:

```
 45s autopkgtest [03:10:08]: test pybuild-autopkgtest: [-----------------------
 46s pybuild-autopkgtest
 46s I: pybuild base:311: cd /tmp/autopkgtest-lxc.w959byms/downtmp/autopkgtest_tmp/build; python3.12 -m pytest 
 47s ============================= test session starts ==============================
 47s platform linux -- Python 3.12.7, pytest-8.3.3, pluggy-1.5.0
 47s rootdir: /tmp/autopkgtest-lxc.w959byms/downtmp/autopkgtest_tmp/build
 47s plugins: typeguard-4.4.1, cov-5.0.0, mock-3.14.0, rerunfailures-14.0
 47s collected 27 items
 47s 
 47s tests/test_cache.py ................s.....                               [ 81%]
 47s tests/test_util.py ...F.                                                 [100%]
 47s 
 47s =================================== FAILURES ===================================
 47s _______________________ test_xor_bytes[\x12\xab-4-&\xab] _______________________
 47s 
 47s b1 = b'\x12\xab', b2 = b'4', r = b'&\xab'
 47s 
 47s     @pytest.mark.parametrize(
 47s         "b1,b2,r",
 47s         [
 47s             (b"\x12", b"\x34", b"\x26"),
 47s             (b"\0\x12", b"\0\x34", b"\0\x26"),
 47s             (b"\x12\0", b"\x34\0", b"\x26\0"),
 47s             (b"\x12\xAB", b"\x34", b"\x26\xAB"),
 47s             (b"\x12\xAB", b"\x34\xCD", b"\x26\x66"),
 47s         ],
 47s     )
 47s     def test_xor_bytes(b1: bytes, b2: bytes, r: bytes) -> None:
 47s >       assert xor_bytes(b1, b2) == r
 47s E       AssertionError: assert b'\x12\x9f' == b'&\xab'
 47s E         
 47s E         At index 0 diff: b'\x12' != b'&'
 47s E         Use -v to get more diff
 47s 
 47s tests/test_util.py:16: AssertionError
 47s =========================== short test summary info ============================
 47s FAILED tests/test_util.py::test_xor_bytes[\x12\xab-4-&\xab] - AssertionError:...
 47s =================== 1 failed, 25 passed, 1 skipped in 0.59s ====================
 47s E: pybuild pybuild:389: test: plugin pyproject failed with: exit code=1: cd /tmp/autopkgtest-lxc.w959byms/downtmp/autopkgtest_tmp/build; python3.12 -m pytest 
 47s pybuild-autopkgtest: error: pybuild --autopkgtest --test-pytest -i python{version} -p 3.12 returned exit code 13
 47s make: *** [/tmp/LMVqFCDGDw/run:4: pybuild-autopkgtest] Error 25
 47s pybuild-autopkgtest: error: /tmp/LMVqFCDGDw/run pybuild-autopkgtest returned exit code 2
 47s autopkgtest [03:10:10]: test pybuild-autopkgtest: -----------------------]
```

To my understanding, it should be the `xor_byte()` function implicitly assuming the little-endian hardware architecture.

The raw test log can be found at https://ci.debian.net/packages/f/fscacher/testing/s390x/54726807/ .

Please let me know how you would like to have this issue fixed, such as skipping the test on big-endians or pushing an actual fix. Thanks!

--- a/src/fscacher/cache.py
+++ b/src/fscacher/cache.py
@@ -6,7 +6,6 @@ import logging
 import os
 import os.path as op
 import shutil
-import sys
 import time
 import joblib
 from platformdirs import PlatformDirs
@@ -247,9 +246,11 @@ class DirFingerprint:
 
 def xor_bytes(b1: bytes, b2: bytes) -> bytes:
     length = max(len(b1), len(b2))
-    i1 = int.from_bytes(b1, sys.byteorder)
-    i2 = int.from_bytes(b2, sys.byteorder)
-    return (i1 ^ i2).to_bytes(length, sys.byteorder)
+    # force 'little' byte order to match our assumptions on how to
+    # treat bytes of different length.
+    i1 = int.from_bytes(b1, "little")
+    i2 = int.from_bytes(b2, "little")
+    return (i1 ^ i2).to_bytes(length, "little")
 
 
 def elapsed_since(t: float) -> float:

--- a/src/fscacher/tests/test_util.py
+++ b/src/fscacher/tests/test_util.py
@@ -1,6 +1,5 @@
 from itertools import zip_longest
 import pytest
-
 from ..cache import xor_bytes
 
 

--- a/src/fscacher/tests/test_util.py
+++ b/src/fscacher/tests/test_util.py
@@ -1,4 +1,6 @@
+from itertools import zip_longest
 import pytest
+
 from ..cache import xor_bytes
 
 
@@ -13,4 +15,6 @@ from ..cache import xor_bytes
     ],
 )
 def test_xor_bytes(b1: bytes, b2: bytes, r: bytes) -> None:
+    # test the target value assumption
+    assert r == bytes(x ^ y for x, y in zip_longest(b1, b2, fillvalue=0))
     assert xor_bytes(b1, b2) == r

